test/common/http/http1/conn_pool_test.cc

```diff
@@ -506,6 +506,25 @@ TEST_F(Http1ConnPoolImplTest, CancelBeforeBound) {
   dispatcher_.clearDeferredDeleteList();
 }
 
+/**
+ * Test cancelling with CloseExcess
+ */
+TEST_F(Http1ConnPoolImplTest, CancelExcessBeforeBound) {
```

Comment:
I think the test would be better if there are at least 2 pending requests, to validate that it only closes the Excess connections, not all connections.

Reply:
If you're OK I'm going to just beef up the http2 tests (some with multiplexing, some without) just because that test fixture is set up better for it and it's testing shared code.